src/Symfony/Component/Form/Util/VirtualFormAwareIterator.php

```diff
@@ -22,11 +20,26 @@
  *
  * @author Bernhard Schussek <bschussek@gmail.com>
  *
- * @deprecated Deprecated since version 2.3, to be removed in 3.0. Use
- *             {@link InheritDataAwareIterator} instead.
+ * @deprecated since version 2.3, to be removed in 3.0.
+ *             Use {@link InheritDataAwareIterator} instead.
  */
 class VirtualFormAwareIterator extends \IteratorIterator implements \RecursiveIterator
 {
+    public function __construct(\Traversable $iterator, $triggerDeprecationNotice = true)
+    {
+        /**
+         * Prevent to trigger deprecation notice when already using the
+         * InheritDataAwareIterator class that extends this deprecated one.
+         * The {@link Symfony\Component\Form\Util\InheritDataAwareIterator::__construct} method
+         * forces this argument to false.
+         */
+        if ($triggerDeprecationNotice) {
```

Comment:
instead of adding an extra argument, it would be more reliable to use ``get_class($this) === __CLASS__`` given that our use case for avoiding the trigger is for the child class.